gato/enumerate/repository.py

```diff
@@ -132,17 +138,7 @@ def enumerate_repository(self, repository: Repository, large_org_enum=False):
             runner_detected = True
 
         if not self.skip_log:
-            # If we are enumerating an organization, only enumerate runlogs if
-            # the workflow suggests a sh_runner.
-            if large_org_enum and runner_detected:
-                self.__perform_runlog_enumeration(repository)
-
-            # If we are doing internal enum, get the logs, because coverage is
-            # more important here and it's ok if it takes time.
-            elif not repository.is_public() and self.__perform_runlog_enumeration(repository):
-                runner_detected = True
-            else:
-                runner_detected = self.__perform_runlog_enumeration(repository)
+            runner_detected = self.__perform_runlog_enumeration(repository)
```

Comment:
```suggestion
            if large_org_enum and runner_detected:
                self.__perform_runlog_enumeration(repository)
            # If we are doing internal enum, get the logs, because coverage is
            # more important here and it's ok if it takes time.
            elif not repository.is_public() and self.__perform_runlog_enumeration(repository):
                runner_detected = True
            elif not large_org_enum:
                runner_detected = self.__perform_runlog_enumeration(repository)
```

Good catch on the logic error, this is really what the logic should be here. Essentially:

* If it's a public repo and we are running on a larger org (> 100 repos), only perform runlog analysis if the workflow static analysis suggests it might be using a self-hosted runner.
* If it's a private repo, perform analysis since this is likely post-compromise enum and we want as much information as possible.
* If it's a smaller org (<100 repos) then get the run logs.

Reply:
This makes a lot more sense! However, this would still cause `self.__perform_runlog_enumeration` to be executed twice in the event that large_org_enum is False since it's running once at the first elif statement and running once again on the second elif statement 

Would something like the following help prevent that?  

```
            if large_org_enum and runner_detected:
                self.__perform_runlog_enumeration(repository)
            # If we are doing internal enum, get the logs, because coverage is
            # more important here and it's ok if it takes time.
            elif not repository.is_public() or not large_org_enum:
                runner_detected = self.__perform_runlog_enumeration(repository)

```